SingularityUI/app/components/common/modalButtons/DeletePendingTaskButton.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { Component, PropTypes } from 'react';
+
+import { Glyphicon } from 'react-bootstrap';
+import OverlayTrigger from 'react-bootstrap/lib/OverlayTrigger';
+import ToolTip from 'react-bootstrap/lib/Tooltip';
+
+import { getClickComponent } from '../modal/ModalWrapper';
+
+import DeletePendingTaskModal from './DeletePendingTaskModal';
+
+const deletePendingTaskTooltip = (
+  <ToolTip id="delete">
+    Delete this pending task
+  </ToolTip>
+);
+
+export default class DeletePendingTaskButton extends Component {
+
+  static propTypes = {
+    taskId: PropTypes.string.isRequired,
+    requestType: PropTypes.string.isRequired,
+    then: PropTypes.func
+  };
+
+  static defaultProps = {
+    children: (
+      <OverlayTrigger placement="top" id="view-delete-pending-task-overlay" overlay={deletePendingTaskTooltip}>
+        <a>
+          <Glyphicon glyph="trash" />
+        </a>
+      </OverlayTrigger>
+    )
+  };
+
+  render() {
+    if (this.props.requestType == "ON_DEMAND") {
```

Comment:
Rather than making this check here, I would only render the button in the first place if the type is `ON_DEMAND`. Then you don't need to pass the `requestType` as a prop at all.

Reply:
Since the `DeletePendingTaskButton` is placed inside the `ScheduledActions` constant, how can I make it dynamic?

Reply:
You could wrap it in a conditional, like 
```
{cellData.pendingTask.pendingTaskId.id &&
    <DeletePendingTaskButton 
        taskId={cellData.pendingTask.pendingTaskId.id}
        requestType={cellData.request.requestType}
    />
}
```